pkg/apis/v1/nodeclaim.go

```diff
@@ -61,6 +61,16 @@ type NodeClaimSpec struct {
 	// If left undefined, the controller will wait indefinitely for pods to be drained.
 	// +optional
 	TerminationGracePeriod *metav1.Duration `json:"terminationGracePeriod,omitempty"`
+	// ExpireAfter is the duration the controller will wait
+	// before terminating a node, measured from when the node is created. This
+	// is useful to implement features like eventually consistent node upgrade,
+	// memory leak protection, and disruption testing.
+	// +kubebuilder:default:="720h"
+	// +kubebuilder:validation:Pattern=`^(([0-9]+(s|m|h))+)|(Never)$`
+	// +kubebuilder:validation:Type="string"
+	// +kubebuilder:validation:Schemaless
+	// +optional
+	ExpireAfter NillableDuration `json:"expireAfter,omitempty"`
```

Comment:
Since this is a field that can be hashed, we should problem add a test for it

Reply:
Just validating, we would like this change to be included in v1, right? if it's not included we will need to make sure that we bump the hash version

Reply:
Good call. Will add it in

Reply:
Yeah it should be in v1